docs/internals.rst

```diff
@@ -1656,6 +1656,40 @@ significant variance. Experience shows that a ~20% time difference can be
 reliably measured on GitHub hosted runners, and smaller differences are
 noticeable over a few days of reruns as trends emerge from the noise.
 
+Code coverage
+-------------
+
+Code coverage for Verilator itself can be collected using GCC with the
+following flow. Note that configuring with ``--enable-coverage`` disables
+optimization for both the debug and optimized builds of Verilator, so running
+the resulting executables can be slow:
+
+.. code:: shell
+
+   ./configure --enable-longtests --enable-coverage
+   make -j$(nproc)        # Build verilator
+   make test              # Run the tests - this will generate .gcda files
+   make coverage-open     # Create and open HTML coverate reprot
+
+
+The ``coverage-open`` make target depends on ``covarage-report``, which
+generates the HTML coverage report, and does in turn depend on
+``coverage-combine``, which combines all ``.gcda`` files into an lcov
+data file. You can use each of these make targets separately if you wish.
+
+If you want to see the coverage only for select test, instead of ``make test``,
+you can just run the relevant ``test_regress/t/*.py`` cases, then use
+``make coverage-open``
+
+You can also use the ``coverage-zero`` target to remove all ``.gcda`` files,
+which in effect clears all collected coverage data. This can be useful when
+checking individual test cases while modifying them.
+
+Note that every time you invoke a binary built with coverage collection, it
+will add the coverage data collected during that invocation to the existing
+``.gcda`` files. This means that you can also just run Verilator on any other
+external files or projects to collect code coverage on those invocations.
+
```

Comment:
Be aware if changing a test, if that test no longer covers some item the report will still contain the old coverage.  Use a make `coverage-zero` and rerun all tests if this is a concern.